### PR TITLE
Issue Fix for OCL

### PR DESCRIPTION
**Pull request by HamedHemati**
- FIX an issue with eval stream initialization
- UPDATE interactive logger for online CL strategies

### CODE REVIEW

File: avalanche/training/templates/base.py [hunk 1]
```diff
@@ -106,7 +106,11 @@ def train(
             experiences = [experiences]
         if eval_streams is None:
             eval_streams = [experiences]
-        self._eval_streams = _group_experiences_by_stream(eval_streams)
+
+        if len(eval_streams) == 1:
+            self._eval_streams = eval_streams
+        else:
```
Comment (AntonioCarta) [inline avalanche/training/templates/base.py:112]: can you put this inside`_group_experiences_by_stream`?

**Comment by AntonioCarta** [comment]
can you add some tests to catch the error?

File: avalanche/training/templates/base.py (hunk 1 above)
Comment (HamedHemati) [reply avalanche/training/templates/base.py:112]: yea, sure.

**Comment by HamedHemati** [comment]
> can you add some tests to catch the error?

What do you mean by adding tests? is there something that needs to be caught?

**Comment by AntonioCarta** [comment]
> What do you mean by adding tests? is there something that needs to be caught?

We need a unit test that fails if the bug with the OCL stream occurs again because of some changes.

**Comment by HamedHemati** [comment]
> We need a unit test that fails if the bug with the OCL stream occurs again because of some changes.

The online strategy test is updated. Now, it checks `assert cl_strategy.clock.train_exp_counter > 0` after training on the whole OCL stream.

**Comment by AntonioCarta** [comment]
Sorry, I didn't explain myself. This PR is fixing a bug:

> FIX an issue with eval stream initialization

If we want to be sure the bug doesn't happen again, we need a unit test that checks this problem, i.e. passes an OCL eval stream and checks that the periodic eval groups the streams correctly in all the possible cases.

**Pull request merged**